### PR TITLE
[browser] legacy JS interop optional via WasmEnableLegacyJsInterop - native

### PR DESCRIPTION
**Pull request by pavelsavara**
Initial work to make trim legacy interop from native code and JavaScript.

Split off from https://github.com/dotnet/runtime/pull/79622

## Changes
- added `WasmEnableLegacyJsInterop` MSBuild feature. 
    - discussion about name [here ](https://github.com/dotnet/runtime/pull/79622#discussion_r1050808657)
    - defaults to true
    - It's available in runtime build as well as in application re-link via workload. At this time they need to be used together.
    - the re-link with `WasmEnableLegacyJsInterop==false` will not make the JS smaller. That would be follow-up work.
    - it maps to `ENABLE_LEGACY_JS_INTEROP` in native code
- moved all legacy native functions to end of `driver.c` under `#ifdef ENABLE_LEGACY_JS_INTEROP`
    - just moved `mono_wasm_get_delegate_invoke_ref` and `mono_wasm_box_primitive_ref` lower in the same file.
    - just moved `mono_wasm_get_type_name` and `mono_wasm_get_type_aqn` higher in the same file.
    - moved `mono_wasm_typed_array_new_ref` from `corebindings.c` to `driver.c`
- renamed `core_initialize_internals` to `bindings_initialize_internals` and unified it with `mono_initialize_internals`
    - removed `CORE_BINDINGS` definition
- moved legacy cwraps to `legacy_c_functions`
    - methods in `cwraps.ts` and `corebindings.c` and `dotnet.es6.lib.js` are sorted in the same order
- made TS type imports more explicit
- moved `mono_method_resolve` to `export_internal_api`
- made `dotnet.es6.lib.js` driven my environment variables, so that it could be configured at dev machine build time
    - see `EmscriptenEnvVars` in `WasmApp.Native.targets`
    - the same logic is also for rollup and runtime build. See `wasm.proj`
- moved `loadedFiles` from 'Module' to `runtimeHelpers`

## Future work
- create multiple builds that the workload re-link could choose from based on the flags. This would make it useful for customer.
- same transformation could be done for jiterpretter

### CODE REVIEW

**Comment by kg** [comment]
Please ensure none of these funcs are used by the jiterpreter (CI ideally will catch it but isn't guaranteed)

**Comment by pavelsavara** [comment]
> Please ensure none of these funcs are used by the jiterpreter (CI ideally will catch it but isn't guaranteed)

@kg
I split method removal to separate PR, in to make it easier to review https://github.com/dotnet/runtime/pull/82865

The rest of here should be just moving them into conditional block. I will re-base this on the other PR later.

**Comment by pavelsavara** [comment]
CI failure is https://github.com/dotnet/runtime/issues/80619

File: src/mono/wasm/runtime/es6/dotnet.es6.lib.js [hunk 1]
```diff
@@ -4,13 +4,9 @@
...
 "use strict";
 
-#if USE_PTHREADS
-const usePThreads = true;
-const isPThread = `ENVIRONMENT_IS_PTHREAD`;
-#else
-const usePThreads = false;
-const isPThread = `false`;
-#endif
+const monoWasmThreads = process.env.MonoWasmThreads == "true";
```
Comment (kg) [inline src/mono/wasm/runtime/es6/dotnet.es6.lib.js:7]: This works because emscripten evaluates dotnet.es6.lib.js in node at build time, right?

**Review by kg — APPROVED** [review]

File: src/mono/wasm/runtime/es6/dotnet.es6.lib.js (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/es6/dotnet.es6.lib.js:7]: Yes

**Pull request merged**